File: src/main/java/org/luxons/sevenwonders/config/AnonymousUsersHandshakeHandler.java
package org.luxons.sevenwonders.config;

import java.security.Principal;
import java.util.Map;

import org.springframework.http.server.ServerHttpRequest;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.web.socket.WebSocketHandler;
import org.springframework.web.socket.server.support.DefaultHandshakeHandler;

class AnonymousUsersHandshakeHandler extends DefaultHandshakeHandler {

    private int playerId = 0;

    @Override
    public Principal determineUser(ServerHttpRequest request, WebSocketHandler wsHandler,
            Map<String, Object> attributes) {
        Principal p = super.determineUser(request, wsHandler, attributes);
        if (p == null) {
            p = new UsernamePasswordAuthenticationToken("player" + playerId++, null);
        }
        return p;
    }
}

File: sw-server/src/main/kotlin/org/luxons/sevenwonders/server/config/AnonymousUsersHandshakeHandler.kt
package org.luxons.sevenwonders.server.config

import org.springframework.http.server.ServerHttpRequest
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.web.socket.WebSocketHandler
import org.springframework.web.socket.server.support.DefaultHandshakeHandler
import java.security.Principal

/**
 * Generates [Principal] objects for anonymous users in the form "playerX", where X is an auto-incremented number.
 */
internal class AnonymousUsersHandshakeHandler : DefaultHandshakeHandler() {

    private var playerId = 0

    override fun determineUser(
        request: ServerHttpRequest,
        wsHandler: WebSocketHandler,
        attributes: Map<String, Any>
    ): Principal? {
        var p = super.determineUser(request, wsHandler, attributes)
        if (p == null) {
            p = UsernamePasswordAuthenticationToken("player" + playerId++, null)
        }
        return p
    }
}
